### PR TITLE
Update to React 16 and move to ES6

### LINKED ISSUE #24: Uncaught Error: addComponentAsRefTo(...): Only a ReactOwner can have refs.
I have this error when I try to use react-progressbar.js in my react App.
error

I figured it was because react was called inside react-progressbar.js
npm ls react

I don't know how to fix it. I tried using alias and putting react in externals but nothing worked... 
Here is my webpack. 
```js
const webpack = require('webpack');
const path = require('path');
module.exports = {
  entry: [
    './src/index.js'
  ],
  output: {
    path: path.join(__dirname, '/public/'),
    publicPath: '/',
    filename: 'bundle.js'
  },
  plugins: [
    new webpack.ProvidePlugin({
      WaveSurfer: 'wavesurfer.js'
    })
  ],
  module: {
    loaders: [{
      exclude: /node_modules/,
      loader: 'babel',
      query: {
        presets: ['react', 'es2015', 'stage-1']
      }
    },
    {test: /\.(jpe?g|png|gif|svg)$/i,
        loaders: [
          'file?hash=sha512&digest=hex&name=[hash].[ext]',
          'image-webpack?bypassOnDebug&optimizationLevel=7&interlaced=false'
        ]}
  ]
  },
  resolve: {
    extensions: ['', '.js', '.jsx'],
    alias: {
      wavesurfer: require.resolve('wavesurfer.js')
    }
  },
  devServer: {
    historyApiFallback: true,
    contentBase: './'
  }
};
```

And here is my package.json
```js
{
  "name": "corleone",
  "version": "1.0.0",
  
  "main": "index.js",
  "repository": "git@gitlab.com:analysR/corleone.git",
  "scripts": {
    "dev": "node ./node_modules/webpack-dev-server/bin/webpack-dev-server.js",
    "build": "webpack --config ./webpack.config.js --progress",
    "start": "node ./server/index.js",
    "test": "mocha --compilers js:babel-core/register --require ./test/test_helper.js --recursive ./test",
    "test:watch": "npm run test -- --watch"
  },
  "author": "Loïc Berthelot & Pierre Jankowiez",
  "license": "ISC",
  "devDependencies": {
    "babel-core": "^6.2.1",
    "babel-loader": "^6.2.0",
    "babel-preset-es2015": "^6.1.18",
    "babel-preset-react": "^6.1.18",
    "chai": "^3.5.0",
    "chai-jquery": "^2.0.0",
    "file-loader": "^0.11.2",
    "image-webpack-loader": "^3.4.2",
    "jquery": "^2.2.1",
    "jsdom": "^8.1.0",
    "mocha": "^2.4.5",
    "react-addons-test-utils": "^0.14.7",
    "webpack": "^1.12.9",
    "webpack-dev-server": "^1.14.0"
  },
  "dependencies": {
    "axios": "^0.16.2",
    "babel-preset-stage-1": "^6.1.18",
    "body-parser": "^1.18.2",
    "bv": "git+https://PierreJankoECE:NpUNsx5Go8FH46GFyAP_@gitlab.com/analysR/bv.git",
    "chain": "git+https://PierreJankoECE:NpUNsx5Go8FH46GFyAP_@gitlab.com/analysR/chain.git",
    "cookie-parser": "^1.4.3",
    "cors": "^2.8.4",
    "express": "^4.15.5",
    "firebase-key": "^1.0.5",
    "lodash": "^3.10.1",
    "morgan": "^1.9.0",
    "multer": "^1.3.0",
    "nconf": "^0.8.5",
    "nedb": "^1.8.0",
    "nocache": "^2.0.0",
    "react": "^0.14.3",
    "react-audio-player": "^0.6.2",
    "react-dom": "^0.14.3",
    "react-file-input": "^0.2.5",
    "react-loader": "^2.4.2",
    "react-notifications": "^1.4.3",
    "react-progressbar.js": "^0.2.0",
    "react-redux": "4.3.0",
    "react-router": "^2.0.1",
    "react-wavesurfer": "^0.8.6",
    "redux": "^3.0.4",
    "redux-promise": "^0.5.3",
    "redux-thunk": "^2.2.0",
    "socket.io": "^2.0.3",
    "vantage": "^1.7.0",
    "vorpal": "^1.12.0",
    "wavesurfer.js": "^1.4.0"
  }
}

```

Thank you very much

### PR DESCRIPTION
**Pull request by sebacruz**
Both updates are related somehow so I put it in the same PR.

Also this should fix #24.

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -1,61 +1,62 @@
...
             shape: null
         };
-    },
+    }
 
-    render: function render() {
-        var style = this.props.containerStyle;
-        var className = this.props.containerClassName;
+    render() {
+        const {style, className} = this.props;
```
Comment (manhhailua) [inline src/main.js:26]: @sebacruz should be `containerStyle` and `containerClassName`
```
const { containerStyle, containerClassName } = this.props;
```

**Comment by Kikobeats** [comment]
Can you merge this @kimmobrunfeldt?

**Comment by isualize** [comment]
I would really appreciate mergin this PR

**Comment by erosenberg** [comment]
Me too!

**Comment by Kikobeats** [comment]
I tried to be contributor of the repository for fix theses necessary PR's but @kimmobrunfeldt is not replying 😢

**Comment by erosenberg** [comment]
@Kikobeats - Bah, that sucks!!
I tried @sebacruz 's repo and ran into errors regarding importing `isEqual` from lodash.
It appears somehow it's looking for a package `isequal` instead of `isEqual`...

**Comment by sebacruz** [comment]
@erosenberg I updated the `isEqual` package name match the lodash/isEqual file, can you test if it's working now?

**Comment by erosenberg** [comment]
@sebacruz - looks good now, thanks!!

**Comment by Kikobeats** [comment]
What do you think create a fork package of this?

**Comment by Jivings** [comment]
I've forked this repo here: https://github.com/squarecat/react-progressbar.js and it's now working in my React 16 project.

I've applied the patch of this PR, and given the repo public push access, so if anyone wants to move over any of the other PRs then feel free.

**Comment by Kikobeats** [comment]
Thanks for your effort @Jivings! 

I'm seeing another error related with [this line](https://github.com/squarecat/react-progressbar.js/blob/master/src/main.js#L28). My code said:

```
Uncaught Error: Element ref was specified as a string (progressBar) but no owner was set. You may have multiple copies of React loaded. (details: https://fb.me/react-refs-must-have-owner).
```

I think this is the correct way to setup a ref node:
https://github.com/kimmobrunfeldt/react-progressbar.js/pull/18/files#diff-1c90ff38a08209f9ebd4d05d1e43358eR79

It's used by the Shape internal components:
https://github.com/kimmobrunfeldt/react-progressbar.js/pull/18/files#diff-1e9094ba61ef643007d82d726a1ade45R149

**Comment by Jivings** [comment]
@Kikobeats Yep you're right. I also had to follow the answer in [this thread](https://github.com/kimmobrunfeldt/react-progressbar.js/issues/24#issuecomment-343487407).

I think this is due to react-progressbar importing it's own version of React so that the version in your code sees the ref as belonging to a different owner.

**Comment by Kikobeats** [comment]
@Jivings hmmm I'm not sure. Probably the code should be write as ES6 using imports and be shipped using babel compiler to ensure module injection is working as expected.

I wrote a package using this pattern, you can check the code here: https://github.com/Kikobeats/react-img-atv

I'm going to try create a PR doing that

**Comment by Jivings** [comment]
Yep I agree. These are just workarounds for a component that hasn't been built to be particularly resilient to it's environment.

**Comment by mssngr** [comment]
Would love to use this library, but my project's React 16. Looking forward to this some day getting merged?

**Comment by Kikobeats** [comment]
@mssngr uses the @Jivings forks, it is working like a charm
https://github.com/squarecat/react-progressbar.js

Could be great release it as `@squarecat/react-progressbar.js`. 

Feel free to use https://www.npmjs.com/package/react-bar-progress as well 🙂

**Comment by jmfurlott** [comment]
@mssngr @Kikobeats in the meantime you guys can just do:
```
npm install git://github.com/squarecat/react-progressbar.js.git
```

Would be nice for a proper merge though as this is fragile from @squarecat

**Comment by Jivings** [comment]
@jmfurlott I actually published the fork to npm so you can do;

```npm install react-progress-bar.js```

@mssngr Unfortunately it looks like this project has been abandoned, but if anyone wants to contribute to the fork then let me know and I'll just grant write access.

**Comment by kimmobrunfeldt** [comment]
I'm sorry that I haven't been able to keep up with the maintenance. If @Jivings or others would be interested to help with this repo, I'm totally open to give access.

File: src/main.js (hunk 1 above)
Comment (abdurrahmanekr) [reply src/main.js:26]: I opened PR for this. #32

**Pull request merged**